src/events/http/HttpServer.js

```diff
@@ -538,12 +539,23 @@ export default class HttpServer {
         const stageVariables = this.#serverless.service.custom
           ? this.#serverless.service.custom.stageVariables
           : null
-        const lambdaProxyIntegrationEvent = new LambdaProxyIntegrationEvent(
-          request,
-          this.#serverless.service.provider.stage,
-          requestPath,
-          stageVariables,
-        )
+        let lambdaProxyIntegrationEvent
+
+        if (endpoint.isHttpApi && endpoint.payload === '2.0') {
+          lambdaProxyIntegrationEvent = new LambdaProxyIntegrationEventV2(
+            request,
+            this.#serverless.service.provider.stage,
+            requestPath,
+            stageVariables,
+          )
+        } else {
+          lambdaProxyIntegrationEvent = new LambdaProxyIntegrationEvent(
+            request,
+            this.#serverless.service.provider.stage,
+            requestPath,
+            stageVariables,
+          )
+        }
```

Comment:
what about?
```
const ProxyClass = (endpoint.isHttpApi && endpoint.payload === '2.0')?LambdaProxyIntegrationEventV2:LambdaProxyIntegrationEvent;
const lambdaProxyIntegrationEvent = new ProxyClass(
  request,
  this.#serverless.service.provider.stage
  requestPath,
  stageVariables,
)
```

That way we can keep the const's?

Reply:
Added back const